### PR TITLE
Remove Python 3.4 run from Travis-CI

### PR DESCRIPTION
**Pull request by pitrou**
This may help speed up builds a bit.
Also remove the ASYNCIO environment variable which doesn't seem examined anywhere.
Also taking the liberty of adding `setup.py` classifiers.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -41,6 +41,19 @@
                 'distributed.http'],
       long_description=(open('README.rst').read() if os.path.exists('README.rst')
                         else ''),
+      classifiers=[
+        "Development Status :: 5 - Production/Stable",
```
Comment (pitrou) [inline setup.py]: Perhaps I'm being optimistic?

Comment (mrocklin) [inline setup.py]: Seems fine to me.  This seems project seems stable relative to most other projects.

File: .travis.yml [hunk 2]
```diff
@@ -7,9 +7,8 @@ services:
 env:
   matrix:
     - PYTHON=2.7 PACKAGES="blosc futures faulthandler"
-    - PYTHON=3.4 COVERAGE=true DASK_EXPERIMENTAL_ZMQ=1
-    - PYTHON=3.5 CRICK=true ASYNCIO=true
-    - PYTHON=3.6 PACKAGES=blosc ASYNCIO=true
+    - PYTHON=3.5 COVERAGE=true PACKAGES=blosc CRICK=true DASK_EXPERIMENTAL_ZMQ=1
```
Comment (mrocklin) [inline .travis.yml]: Thoughts on dropping ZMQ?

Comment (pitrou) [inline .travis.yml]: No hard feelings from me. It we drop testing it, though, we can probably drop it from the code base as well. What do you think?

Comment (mrocklin) [inline .travis.yml]: I think it's value is in verifying that comms is extensible.  However now that `inproc` exists that may suffice.  I'm inclined to remove it just to reduce the codebase.

**Comment by mrocklin** [comment]
Merged

**Pull request merged**